### PR TITLE
Feature/rheadley/bugfix for prenaffl

### PR DESCRIPTION
**Pull request by lifewithryan**
# Critical Changes

# Changes

- added ability for custom picklist component to optionally include '--None--' as an option in order to allow blank values to be set.

# Issues Closed
- Fixes [W-030890](https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000007fQxsQAE/view)

# New Metadata

- New CustomLabel call "NoneOption" with a value of '--None--'

# Deleted Metadata

# Testing Notes
- Ensure HEDA Settings -> Affiliations -> Settings tab shows a selected option when in view mode (including when it is set to --None--)
- Create Program Enrollment on a Contact
- Verify Affiliation gets created according to specified settings
- Delete Program Enrollment
- Verify Affiliation is deleted or Status of Affiliation is updated according to settings.

### CODE REVIEW

**Comment by lifewithryan** [comment]
@xinnanlisf I cleand up some tabs to spaces stuff in some of the JS files so it looks like more than what is actually there :)

File: src/aura/STG_CMP_Affl/STG_CMP_AfflHelper.js [hunk 1]
```diff
@@ -1,101 +1,108 @@
 ({
```
Comment (xinnanlisf) [inline src/aura/STG_CMP_Affl/STG_CMP_AfflHelper.js]: Hi @lifewithryan, I did some eyeball comparison but just want to make sure the only change you made in this class is format change.

File: src/aura/STG_CMP_Tabs/STG_CMP_TabsHelper.js [hunk 2]
```diff
@@ -138,26 +143,35 @@
...
                 var picklistValues = response.getReturnValue();
                 var affiliationStatusPicklistEntries = [];
+        
                 for(var property in picklistValues) {
+                    
                     if (picklistValues.hasOwnProperty(property)) {
                         affiliationStatusPicklistEntries.push({picklistLabel: property, picklistValue: picklistValues[property]});
-                        if(property.indexOf(statusValue) > -1) {
+                        
+                        //if(property.indexOf(statusValue) > -1 || property == '--None--') {
```
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/aura/STG_CMP_Tabs/STG_CMP_TabsHelper.js]: Yes, that's on purpose and I must have forgotten to remove it. The code no longer uses --None-- as I have move that text to a label.

File: src/aura/STG_CMP_Affl/STG_CMP_AfflHelper.js (hunk 1 above)
Comment (lifewithryan) [reply src/aura/STG_CMP_Affl/STG_CMP_AfflHelper.js]: Correct, purely cosmetic at this point. Originally there were changes but after moving all of the picklist building out of the backend and into the component itself, I no longer needed it where it was, so everything left here is my cosmetic stuff.

File: src/aura/STG_CMP_Tabs/STG_CMP_TabsHelper.js [hunk 3]
```diff
@@ -138,26 +143,35 @@
         var prefix = component.get("v.namespacePrefix");
         var objectName = prefix + 'Affiliation__c';
         var fieldName = prefix + 'Status__c';
+
         action.setParams({ "objectName" : objectName,"fieldName" : fieldName});
         action.setCallback(this, function(response) {
             if(response.getState() === "SUCCESS") {
                 var picklistValues = response.getReturnValue();
                 var affiliationStatusPicklistEntries = [];
+
```
[the remark this answers is not in the document]
Comment (lifewithryan) [reply src/aura/STG_CMP_Tabs/STG_CMP_TabsHelper.js]: I do this for readability I like the use of whitespace. I can put it back if you thing moving forward we shouldn't do such things.

**Comment by dgamboni** [comment]
Hi @dashattuck @lifewithryan -- I'm seeing that the "Feature Test - Robot" has failed tests. Do these need to pass before I may merge the branch?

**Comment by dashattuck** [comment]
They need to pass.  It looks like the build is runnng again.

**Pull request merged**